2021/day15/day15.java

```diff
@@ -0,0 +1,116 @@
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.PriorityQueue;
+import java.util.Scanner;
+
+public class day15 {
+    public static class Point {
+        int x;
+        int y;
+
+        Point(int x, int y) {
+            this.x = x;
+            this.y = y;
+        }
+    }
+
+    public static class Position extends Point implements Comparable<Position> {
+        int risk;
+
+        Position(int x, int y, int risk) {
+            super(x, y);
+            this.risk = risk;
+        }
+
+        @Override
+        public int compareTo(Position other) {
+            return Integer.compare(this.risk, other.risk);
+        }
+    }
+
+    public static ArrayList<ArrayList<Integer>> readInput(String filename) {
+        ArrayList<ArrayList<Integer>> mapOfRisks = new ArrayList<>();
+        Scanner sc;
+        try {
+            sc = new Scanner(new File(filename));
+            while (sc.hasNextLine()) {
+                mapOfRisks.add(splitRow(sc.nextLine()));
+            }
+            sc.close();
+        } catch (Exception e) {
+            System.out.println("Error handling input " + e);
+        }
+        return mapOfRisks;
+    }
+
+    public static ArrayList<Integer> splitRow(String row) {
+        ArrayList<Integer> rowOfRisks = new ArrayList<>();
+        for (String s : new ArrayList<String>(Arrays.asList(row.split("")))) {
+            rowOfRisks.add(Integer.parseInt(s));
+        }
+        return rowOfRisks;
+    }
+
+    public static int getRisk(ArrayList<ArrayList<Integer>> mapOfRisks, int x, int y) {
+        int tmpRisk = mapOfRisks.get(y % mapOfRisks.size()).get(x % mapOfRisks.get(0).size())
+                + (x / mapOfRisks.get(0).size()) + (y / mapOfRisks.size());
+        return tmpRisk < 10 ? tmpRisk : ((tmpRisk % 10) + 1);
```

Comment:
This is unreadable. I would extract the `width` and the `height` as parameters before line 57. Also, the last two terms do not need parentheses, I think they make it even more convoluted.

Finally the last line could just be an `if` for readability given the somewhat complicated `else` case